Xamarin.Forms.Build.Tasks/TypeReferenceExtensions.cs

```diff
@@ -82,44 +105,54 @@ public static bool ImplementsInterface(this TypeReference typeRef, TypeReference
 			return false;
 		}
 
+		static readonly string[] arrayInterfaces = {
+			"System.ICloneable",
+			"System.Collections.IEnumerable",
+			"System.Collections.IList",
+			"System.Collections.ICollection",
+			"System.Collections.IStructuralComparable",
+			"System.Collections.IStructuralEquatable",
+		};
+
+		static readonly string[] arrayGenericInterfaces = {
+			"System.Collections.Generic.IEnumerable`1",
+			"System.Collections.Generic.IList`1",
+			"System.Collections.Generic.ICollection`1",
+			"System.Collections.Generic.IReadOnlyCollection`1",
+			"System.Collections.Generic.IReadOnlyList`1",
+		};
+
 		public static bool InheritsFromOrImplements(this TypeReference typeRef, TypeReference baseClass)
 		{
-			var arrayInterfaces = new[]
-			{
-				"System.IEnumerable",
-				"System.Collections.IList",
-				"System.Collections.Collection"
-			};
+			if (TypeRefComparer.Default.Equals(typeRef, baseClass))
+				return true;
 
-			var arrayGenericInterfaces = new[]
-			{
-				"System.IEnumerable`1",
```

Comment:
only existed in my mind. and the mind of the ones reviewing and merging this code